data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportPartitionWorker.java

```diff
@@ -237,12 +247,22 @@ public void run() {
             } catch (Exception e) {
                 LOG.error("Exception connecting to cluster and loading partition {}.", query, e);
                 throw new RuntimeException(e);
-            } finally {
+            }  finally {
                 // Do final checkpoint when reaching end of partition or due to exception
                 partitionCheckpoint.checkpoint(recordCount);
             }
 
             LOG.info("Records processed: {}, recordCount: {}", totalRecords, recordCount);
+        } catch (final IllegalArgumentException | MongoClientException e) {
```

Comment:
The export worker catches this exception and logs it and gives up partition.. New node/thread will acquire the partition and keep retrying. How the aggregate metrics point of view how do we want to handle such error scenarios ?

Reply:
I feel it's OK. Those metrics tracks errors when connecting to DocDB and querying DocDB. They are not indication of export/stream failures.

Reply:
If it keeps retry, errors count keeps incrementing, the invocation count increments as well. And we can use errors over invocations ratio for setting alarms.